qa/tasks/cephfs/test_volumes.py

```diff
@@ -8165,6 +8165,55 @@ def test_unparseable_metafile_on_legacy_to_v1_upgrade(self):
         # remove group
         self._fs_cmd("subvolumegroup", "rm", self.volname, group)
 
+    def test_dangling_symlink(self):
+        """
+        Tests for the presence of any dangling symlink, if yes,
+        will remove it.
+        """
+        subvolume = self._gen_subvol_name()
+        snapshot = self._gen_subvol_snap_name()
+        clone = self._gen_subvol_clone_name()
+
+        # create subvolume in group
+        self._fs_cmd("subvolume", "create", self.volname, subvolume)
+
+        # create subvolume snapshot
+        self._fs_cmd("subvolume", "snapshot", "create", self.volname, subvolume, snapshot)
+
+        # insert delay at the beginning of snapshot clone
+        self.config_set('mgr', 'mgr/volumes/snapshot_clone_delay', 15)
+
+        # disable "capped" clones
+        self.config_set('mgr', 'mgr/volumes/snapshot_clone_no_wait', False)
+
+        # clone
+        self._fs_cmd("subvolume", "snapshot", "clone", self.volname, subvolume, snapshot, clone)
+
+        # list snapshot info
+        result = json.loads(self._fs_cmd("subvolume", "snapshot", "info", self.volname, subvolume, snapshot))
+
+        # verify snapshot info
+        self.assertEqual(result['has_pending_clones'], "yes")
+
+        # remove clone
+        self._fs_cmd("subvolume", "rm", self.volname, clone, "--force")
```

Comment:
Did you test this ? I believe, we no longer allow a inprogess clone to be deleted even with `force` option. We need to cancel the clone before force deletion, that would take care of removing the symlink. 

Please test this out. I think you need to tweak things in the backend to reproduce this issue in latest versions. One way is to directly do rm -rf on the ongoing subvolume clone directory.

Reply:
I see, test require changes.